src/test/java/org/broadinstitute/hellbender/engine/ArgumentsBuilderWithJIMFS.java

```diff
@@ -0,0 +1,146 @@
+        jimfs = Jimfs.newFileSystem(Configuration.unix());
+        jimfsRootPath = jimfs.getRootDirectories().iterator().next();
+        tempDir = jimfsRootPath.resolve(TEMP_DIR);
+        Files.createDirectory(tempDir);
+    }
+
+    @Override
+    public void close() throws IOException {
+        jimfs.close();
+    }
+
+    /**
+     * add a File as a Path/URI by first copying it to JIMFS and then using the resulting Path as the arg value
+     */
+    public ArgumentsBuilder addInputFromJIMFS(final File inputFile) throws IOException {
+        Utils.nonNull(inputFile);
+
+        add("--" + StandardArgumentDefinitions.INPUT_LONG_NAME);
+        final Path path = copyAndResolve(inputFile);
+        add(path.toUri().toString());
+
+        return this;
+    }
+
+    /**
+     * add a File as a Path/URI by first copying it and it's companion index file to JIMFS and then using the
+     * resulting the resulting Path URI as the argument value
+     */
+    public ArgumentsBuilder addInputWithIndexFromJIMFS(final File inputFile, final File indexFile) throws IOException {
+        Utils.nonNull(inputFile);
+        Utils.nonNull(indexFile);
+
+        add("--" + StandardArgumentDefinitions.INPUT_LONG_NAME);
+        final Path path = copyAndResolve(inputFile);
+        copyAndResolve(indexFile);
+        add(path.toUri().toString());
+
+        return this;
+    }
+
+    /**
+     * add a File as a Path/URI by first copying it to JIMFS and then using the resulting Path URI
+     * as the argument value
+     */
+    public ArgumentsBuilder addReferenceFromJIMFS(final File referenceFile) throws IOException {
+        Utils.nonNull(referenceFile);
+
+        // copy the reference, reference dictionary, and reference index
+        final Path path = copyAndResolve(referenceFile);
+        copyAndResolve(ReferenceSequenceFileFactory.getDefaultDictionaryForReferenceSequence(referenceFile));
+        copyAndResolveToName(ReferenceSequenceFileFactory.getFastaIndexFileName(referenceFile.toPath()).toFile(),
+                referenceFile.getName() + ".fai");
+
+        add("--" + StandardArgumentDefinitions.REFERENCE_LONG_NAME);
+        add(path.toUri().toString());
+
+        return this;
+    }
+
+    // TODO: This violates the builder pattern since it needs to return the Path so the test can use it to validate output.
+    public Path addOutputFromJIMFS(final String requestedName) throws IOException {
+        final Path tempOut = getTempFileFromJIMFS(requestedName);
+        add("--" + StandardArgumentDefinitions.OUTPUT_LONG_NAME);
+        add(tempOut.toUri().toString());
+        return tempOut;
+    }
+
+    /**
+     * Obtain a temporary Path to be used as an output file.
+     * @param requestedName name of the requested output file
+     * @return Path of the newly created file in the jimfs temp directory
+     */
+    public Path getTempFileFromJIMFS(final String requestedName) throws IOException {
```

Comment:
Use `Files.createTempFile()`.

Also, can you have the client call `getTempFileFromJIMFS()`themselves, pass the return value in to `addOutputFromJIMFS()`, and have `addOutputFromJIMFS()` return an `ArgumentsBuilder`